File: 977. Squares of a Sorted Array/977.go
package main

import(
	"fmt";
	"math";
	"sort"
)

func main(){
	fmt.Println(sortedSquares([]int{-7,-3,2,3,11}))
}

func sortedSquares(A []int) []int {
	l, r := 0, len(A)-1
	var B []int
	for l<=r{
		if math.Abs(float64(A[l]))>math.Abs(float64(A[r])){
			B = append([]int{A[l]*A[l]}, B...)
			l++
		} else{
			B = append([]int{A[r]*A[r]}, B...)
			r--
		}
	}
	return B
}

func sortedSquares1(A []int) []int {
    B := []int{}
    for _,v := range(A){
        B = append(B,v*v)
    }
    sort.Ints(B)
    return B
}

File: 977. Squares of a Sorted Array/977.py
class Solution:
    def sortedSquares(self, A):
        return sorted([i**2 for i in A])

    def sortedSquares1(self, A):
        answer = [0] * len(A)
        l, r = 0, len(A) - 1
        while l <= r:
            left, right = abs(A[l]), abs(A[r])
            if left > right:
                answer[r - l] = left * left
                l += 1
            else:
                answer[r - l] = right * right
                r -= 1
        return answer

a = Solution()
print(a.sortedSquares( [-4,-1,0,3,10]))